stdlib/runtime/runtimeDebug.gr

```diff
@@ -0,0 +1,712 @@
+            "; new count: " ++
+            toString(newRefCount) ++
+            "; tag: " ++
+            toString(tag) ++
+            " " ++
+            tagToTypeName(tag) ++
+            ")" ++
+            maybeObjectContentsAppendix(userPtr),
+          )
+        } else {
+          if (!Map.contains(userPtr, inintialRefCountsOnPreExistingObjects)) {
+            Map.set(userPtr, oldRefCount, inintialRefCountsOnPreExistingObjects)
+          }
+
+          incrementCountInMap(userPtr, decRefsOnPreExistingObjects)
+
+          print(
+            "decref for pre-existing object at 0x" ++ toHex(userPtr) ++
+            " (prev count: " ++
+            toString(oldRefCount) ++
+            "; new count: " ++
+            toString(newRefCount) ++
+            "; tag: " ++
+            toString(tag) ++
+            " " ++
+            tagToTypeName(tag) ++
+            ")" ++
+            maybeObjectContentsAppendix(userPtr),
+          )
+        }
+      }
+    }
+  }
+
+  let onMalloc = (userPtr: Number) => {
+    totalMallocs += 1
+
+    // Tracking objects allocated in the debug block.
+    if (Set.contains(userPtr, allocations)) {
+      // A new allocation, but for an object already allocated inside the debug block, but never removed.
+      print(
+        "⚠ PROBLEM DETECTED: unexpected malloc at address 0x" ++
+        toHex(userPtr) ++
+        " that hasn't been freed. This may be caused by direct calls to Malloc.free for objects created with GC.malloc.",
+      )
+    } else {
+      // A new allocation made in the debug block.
+      if (options.printMallocAndFree) {
+        print("new object at 0x" ++ toHex(userPtr))
+      }
+
+      // Add the pointer to the new object to start tracking it until it's deallocated.
+      Set.add(userPtr, allocations)
+    }
+  }
+
+  let onFree = (userPtr: Number) => {
+    totalFrees += 1
+
+    let refCount = readRefCount(userPtr)
+    let tag = readTag(userPtr)
+
+    if (Set.contains(userPtr, allocations)) {
+      if (options.printMallocAndFree) {
+        print(
+          "deallocating object at 0x" ++ toHex(userPtr) ++ " (ref. count: " ++
+          toString(refCount) ++
+          "; tag: " ++
+          toString(tag) ++
+          " " ++
+          tagToTypeName(tag) ++
+          ")" ++
+          maybeObjectContentsAppendix(userPtr),
+        )
+      }
+
+      Set.remove(userPtr, allocations)
+
+      incrementCountInMap(tag, mallocPlusFreeByTag)
+    } else {
+      if (options.printMallocAndFree) {
+        print(
+          "deallocating pre-existing object at 0x" ++ toHex(userPtr) ++
+          " (ref. count: " ++
+          toString(refCount) ++
+          "; tag: " ++
+          toString(tag) ++
+          " " ++
+          tagToTypeName(tag) ++
+          ")" ++
+          maybeObjectContentsAppendix(userPtr),
+        )
+      }
+    }
+
+    if (Map.contains(userPtr, incRefsOnPreExistingObjects) ||
+    Map.contains(userPtr, decRefsOnPreExistingObjects)) {
```

Comment:
Is this the formatter output? If so, cc @marcusroberts (I think that `Map` should be indented here)

Reply:
Yes, it is.